src/fsharp/FSharp.Core/printf.fs

```diff
@@ -157,598 +224,238 @@ module internal PrintfImpl =
                         go (i + 1) buf
             go i (Text.StringBuilder())
 
+    [<NoComparison; NoEquality>]
+    /// Represents one step in the execution of a format string
+    type Step =
+        | StepWithArg of prefix: string * conv1: (obj -> string) 
+        | StepWithTypedArg of prefix: string * conv1: (obj -> Type -> string) 
+        | StepString of prefix: string 
+        | StepLittleT of prefix: string 
+        | StepLittleA of prefix: string
+        | StepStar1 of prefix: string * conv: (obj -> int -> string) 
+        | StepPercentStar1 of prefix: string
+        | StepStar2 of prefix: string * conv: (obj -> int -> int -> string)
+        | StepPercentStar2 of prefix: string
+
+        static member BlockCount(steps) =
+            let mutable count = 0
+            for step in steps do 
+                match step with 
+                | StepWithArg (prefix, _conv) ->
+                    if not (String.IsNullOrEmpty prefix) then count <- count + 1
```

Comment:
does whitespace also matter here, or is NullOrEmpty sufficient?

Reply:
We would want to emit whitespace, it would be part of the output string